mapfishapp/src/main/webapp/app/addons/notes/js/main.js

```diff
@@ -0,0 +1,149 @@
+        } else {
+            // addon places in "tools menus"
+            this.item = new Ext.menu.CheckItem({
+                xtype: "button",
+                text: this.getText(record),
+                qtip: this.getQtip(record),
+                icon: GEOR.config.PATHNAME + "/ws/addons/" +
+                    record.get("name").toLowerCase() + "/" +
+                    this.options.icon,
+                checked: false,
+                toggleGroup: this.toggleGroup,
+                listeners: {
+                    "checkchange": this.onToggle,
+                    scope: this
+                }
+            });
+        }
+    },
+    /**
+     * Method: onToggle
+     *
+     */
+    onToggle: function(btn, pressed) {
+        if (pressed) {
+            this.map.events.register("click", this, this.addNote);
+        } else {
+            this.map.events.unregister("click", this, this.addNote);
+        }
+    },
+    /**
+     * Method: addNote
+     *
+     */
+    addNote: function(evt) {
+        var lonlat = this.map.getLonLatFromViewPortPx(evt.xy);
+        lonlat.transform(this.map.projection,
+            new OpenLayers.Projection("EPSG:4326"));
+        this.window = new Ext.Window({
+            title: this.tr("notes_title"),
+            width: 420,
+            closable: true,
+            closeAction: "hide",
+            resizable: false,
+            border: false,
+            items: [{
+                xtype: "form",
+                width: 410,
+                items: [
+                    {
+                        xtype: "textfield",
+                        fieldLabel: this.tr("notes_email"),
+                        width: 240,
+                        name: "email",
+                        allowBlank: false,
+                        value: GEOR.config.USEREMAIL || ""
+                    }, {
+                        xtype: "textarea",
+                        width: 240,
+                        height: 120,
+                        fieldLabel: this.tr("notes_comment"),
+                        name: "comment",
+                        allowBlank: false
+                    }, {
+                        xtype: "hidden",
+                        name: "latitude",
+                        value: lonlat.lat
+                    }, {
+                        xtype: "hidden",
+                        name: "longitude",
+                        value: lonlat.lon
+                    }, {
+                        xtype: "hidden",
+                        name: "map_context",
+                        value: GEOR.wmc.write()
+                    }
+                ],
+                buttons: [{
+                    text: this.tr("notes_submit"),
+                    handler: function(b) {
+                        if (b.findParentByType("form").getForm().isValid()) {
+                            b.findParentByType("form").getForm().submit({
+                                url: GEOR.config.PATHNAME +
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["GEOR" is not defined.](https://www.codacy.com/app/georchestra/georchestra/file/2331320306/issues/source?bid=3200181&fileBranchId=3206983#l109)